### PR TITLE
[DataStorage]Replace getDeviceID functions in storage.go file with helper.go

### PR DESCRIPTION
**Pull request by Seunghui98**
Signed-off-by: Seughui98 <hithere1012@naver.com>

# Description

We have to replace getDeviceID in storage.go file with GetDevice function. 
So, I called GetDevice() in the helper.go file and changed it.

Fixes [#321](https://github.com/lf-edge/edge-home-orchestration-go/issues/321) (issue)

## Type of change

Please delete options that are not relevant.

- [ x ] Code cleanup/refactoring

# How Has This Been Tested?
I took the logs to test and here is the result.
```
INFO[2021-11-02T12:31:13Z]discovery.go:744 clearAllDeviceInfo [discoverymgr] Delete All Device Info        
INFO[2021-11-02T12:31:13Z]discovery.go:839 deleteDevice [discoverymgr] [deleteDevice] edge-orchestration-e61b3b31-9723-47fd-bd49-38aa865e057f 
INFO[2021-11-02T12:31:13Z]networkhelper.go:194 setAddrInfo [networkmgr] addr 10.0.2.15/24               
INFO[2021-11-02T12:31:13Z]discovery.go:398 setDeviceID [discoverymgr] UUID :  e61b3b31-9723-47fd-bd49-38aa865e057f 
INFO[2021-11-02T12:31:13Z]discovery.go:542 SetNetwotkArgument [discoverymgr] [10.0.2.15]                   
INFO[2021-11-02T12:31:13Z]discovery.go:573 func1 [deviceDetectionRoutine] edge-orchestration-e61b3b31-9723-47fd-bd49-38aa865e057f 
INFO[2021-11-02T12:31:13Z]discovery.go:574 func1 [deviceDetectionRoutine] confInfo    : ExecType(container), Platform(docker) 
INFO[2021-11-02T12:31:13Z]discovery.go:575 func1 [deviceDetectionRoutine] netInfo     : IPv4([10.0.2.15]), RTT(0) 
INFO[2021-11-02T12:31:13Z]discovery.go:576 func1 [deviceDetectionRoutine] serviceInfo : Services([]) 
INFO[2021-11-02T12:31:13Z]discovery.go:577 func1                                              
INFO[2021-11-02T12:31:13Z]storage.go:107 StartStorage [storagemgr][deviceName]edge-orchestration-e61b3b31-9723-47fd-bd49-38aa865e057f 
INFO[2021-11-02T12:31:13Z]discovery.go:647 mdnsTXTSizeChecker [discoverymgr] [mdnsTXTSizeChecker] size ::  29  Bytes 
INFO[2021-11-02T12:31:13Z]configuremgr.go:149 Watch start watching for/var/edge-orchestration/apps 
INFO[2021-11-02T12:31:13Z]route.go:84 Add {APIV1Ping GET /api/v1/ping 0xbefe60}        
INFO[2021-11-02T12:31:13Z]route.go:84 Add {APIV1ServicemgrServicesPost POST /api/v1/servicemgr/services 0xbeff00} 
INFO[2021-11-02T12:31:13Z]route.go:84 Add {APIV1ServicemgrServicesNotificationServiceIDPost POST /api/v1/servicemgr/services/notification/{serviceid} 0xbeff80} 
```

**Test Configuration**:
* Firmware version: Ubuntu 20.04
* Hardware: x86-64
* Toolchain: Docker 20.10.6 and Go 1.16.3
* Edge Orchestration Release: Coconut

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by t25kim — CHANGES_REQUESTED** [review]
Thanks for the contribution. However it doesn't work... 
Please take a look my comment! 
Also I would appreciate it if you describe the tests that you ran to verify your changes in `How Has This Been Tested?` from next time!

File: internal/controller/storagemgr/storage.go [hunk 1]
```diff
@@ -60,13 +61,14 @@ var (
...
+	dbIns      dbhelper.MultipleBucketQuery
 	helper     resthelper.RestHelper
 	log        = logmgr.GetInstance()
 )
 
 func init() {
-	deviceID, _ := getDeviceID()
-	deviceName = "edge-orchestration-" + deviceID
+	dbIns = dbhelper.GetInstance()
+	deviceName, _ = dbIns.GetDeviceID()
```
Comment (t25kim) [inline internal/controller/storagemgr/storage.go:71]: It seems that an empty string is assigned to deviceName. 
Please see the log below from `docker logs -f edge-orchestration`.

```
INFO[2021-10-31T21:53:21Z]discovery.go:573 func1 [deviceDetectionRoutine] edge-orchestration-c1b23cc6-0767-400a-9cf0-36e1b3902da2
INFO[2021-10-31T21:53:21Z]discovery.go:574 func1 [deviceDetectionRoutine] confInfo    : ExecType(container), Platform(docker)
INFO[2021-10-31T21:53:21Z]discovery.go:575 func1 [deviceDetectionRoutine] netInfo     : IPv4([10.113.70.227]), RTT(0)
INFO[2021-10-31T21:53:21Z]discovery.go:576 func1 [deviceDetectionRoutine] serviceInfo : Services([DataStorage hello-world])
INFO[2021-10-31T21:53:21Z]discovery.go:577 func1
level=INFO ts=2021-10-31T21:53:21.264525825Z app=datastorage source=config.go:193 msg="Loaded configuration from ./res/configuration.toml"
level=INFO ts=2021-10-31T21:53:21.265237107Z app=datastorage source=config.go:304 msg="Using local configuration from file (0 envVars overrides applied)"
level=INFO ts=2021-10-31T21:53:21.265282604Z app=datastorage source=httpserver.go:99 msg="Web server starting (10.113.70.227:49986)"
level=INFO ts=2021-10-31T21:53:21.26535262Z app=datastorage source=init.go:144 msg="Check Metadata service's status by ping..."
level=INFO ts=2021-10-31T21:53:21.265365384Z app=datastorage source=init.go:144 msg="Check Data service's status by ping..."
level=INFO ts=2021-10-31T21:53:21.266255328Z app=datastorage source=init.go:68 msg="Service clients initialize successful."
level=INFO ts=2021-10-31T21:53:21.270600222Z app=datastorage source=service.go:224 msg="Addressable datastorage exists, updating it"
level=DEBUG ts=2021-10-31T21:53:21.272104648Z app=datastorage source=service.go:155 msg="Trying to find DeviceService: datastorage"
level=INFO ts=2021-10-31T21:53:21.274000129Z app=datastorage source=service.go:177 msg="DeviceService datastorage exists, updating it"
INFO[2021-10-31T21:53:21Z]storagedriver.go:44 Initialize [storagedriver] Device service intialize started
level=DEBUG ts=2021-10-31T21:53:21.283672255Z app=datastorage source=restrouter.go:119 route=/api/v1/device/{deviceName}/resource/{resourceName} methods="[POST GET]" msg="Route added"
level=DEBUG ts=2021-10-31T21:53:21.283754783Z app=datastorage source=restrouter.go:119 route=/api/v1/resource/{resourceName} methods="[POST GET]" msg="Route added"
level=DEBUG ts=2021-10-31T21:53:21.283876825Z app=datastorage source=restrouter.go:119 route=/api/v1/device/{deviceName}/resource/{resourceName}/{readingCount} methods=[GET] msg="Route added"
level=DEBUG ts=2021-10-31T21:53:21.283943766Z app=datastorage source=restrouter.go:119 route=/api/v1/resource/{resourceName}/{readingCount} methods=[GET] msg="Route added"
level=DEBUG ts=2021-10-31T21:53:21.284082732Z app=datastorage source=restrouter.go:119 route=/api/v1/start/{start}/end/{end}/{readingCount} methods=[GET] msg="Route added"
level=INFO ts=2021-10-31T21:53:21.28443934Z app=datastorage source=restrouter.go:75 msg="Registering v2 routes..."
level=DEBUG ts=2021-10-31T21:53:21.284818104Z app=datastorage source=profiles.go:47 msg="created absolute path for loading pre-defined Device Profiles: /edge-orchestration/res"
level=ERROR ts=2021-10-31T21:53:21.287120024Z app=datastorage source=profiles.go:96 msg="Add Device Profile /edge-orchestration/res/datastorage-device.yaml to Core Metadata failed: 400 - Device ID and Name are both blank\n"
level=DEBUG ts=2021-10-31T21:53:21.287155463Z app=datastorage source=devices.go:29 msg="Loading pre-define Devices from configuration"
level=DEBUG ts=2021-10-31T21:53:21.287174423Z app=datastorage source=devices.go:35 msg="Device  doesn't exist, creating a new one"
level=ERROR ts=2021-10-31T21:53:21.287191902Z app=datastorage source=devices.go:58 msg="Device Profile  doesn't exist for Device "
level=ERROR ts=2021-10-31T21:53:21.28720775Z app=datastorage source=devices.go:42 msg="creating Device  from config failed"
level=ERROR ts=2021-10-31T21:53:21.287230532Z app=datastorage source=init.go:96 msg="Failed to create the pre-defined devices: Device Profile  doesn't exist for Device \n"
```

Comment (t25kim) [reply internal/controller/storagemgr/storage.go:71]: The normal log should have the string of the device name like edge-orchestration-c1b23cc6-0767-400a-9cf0-36e1b3902da2 as shown below.

```
INFO[2021-10-31T22:32:31Z]discovery.go:573 func1 [deviceDetectionRoutine] edge-orchestration-c1b23cc6-0767-400a-9cf0-36e1b3902da2
INFO[2021-10-31T22:32:31Z]discovery.go:574 func1 [deviceDetectionRoutine] confInfo    : ExecType(container), Platform(docker)
INFO[2021-10-31T22:32:31Z]discovery.go:575 func1 [deviceDetectionRoutine] netInfo     : IPv4([10.113.70.227]), RTT(0)
INFO[2021-10-31T22:32:31Z]discovery.go:576 func1 [deviceDetectionRoutine] serviceInfo : Services([DataStorage hello-world])
INFO[2021-10-31T22:32:31Z]discovery.go:577 func1
level=INFO ts=2021-10-31T22:32:31.706867391Z app=datastorage source=config.go:193 msg="Loaded configuration from ./res/configuration.toml"
level=INFO ts=2021-10-31T22:32:31.707768328Z app=datastorage source=config.go:304 msg="Using local configuration from file (0 envVars overrides applied)"
level=INFO ts=2021-10-31T22:32:31.707814008Z app=datastorage source=httpserver.go:99 msg="Web server starting (10.113.70.227:49986)"
level=INFO ts=2021-10-31T22:32:31.707883215Z app=datastorage source=init.go:144 msg="Check Metadata service's status by ping..."
level=INFO ts=2021-10-31T22:32:31.708058133Z app=datastorage source=init.go:144 msg="Check Data service's status by ping..."
level=INFO ts=2021-10-31T22:32:31.708938803Z app=datastorage source=init.go:68 msg="Service clients initialize successful."
level=INFO ts=2021-10-31T22:32:31.710402329Z app=datastorage source=service.go:224 msg="Addressable datastorage exists, updating it"
level=DEBUG ts=2021-10-31T22:32:31.712092341Z app=datastorage source=service.go:155 msg="Trying to find DeviceService: datastorage"
level=INFO ts=2021-10-31T22:32:31.713539132Z app=datastorage source=service.go:177 msg="DeviceService datastorage exists, updating it"
INFO[2021-10-31T22:32:31Z]storagedriver.go:44 Initialize [storagedriver] Device service intialize started
level=DEBUG ts=2021-10-31T22:32:31.721246794Z app=datastorage source=restrouter.go:119 route=/api/v1/device/{deviceName}/resource/{resourceName} methods="[POST GET]" msg="Route added"
level=DEBUG ts=2021-10-31T22:32:31.72134553Z app=datastorage source=restrouter.go:119 route=/api/v1/resource/{resourceName} methods="[POST GET]" msg="Route added"
level=DEBUG ts=2021-10-31T22:32:31.721481438Z app=datastorage source=restrouter.go:119 route=/api/v1/device/{deviceName}/resource/{resourceName}/{readingCount} methods=[GET] msg="Route added"
level=DEBUG ts=2021-10-31T22:32:31.721603163Z app=datastorage source=restrouter.go:119 route=/api/v1/resource/{resourceName}/{readingCount} methods=[GET] msg="Route added"
level=DEBUG ts=2021-10-31T22:32:31.72174911Z app=datastorage source=restrouter.go:119 route=/api/v1/start/{start}/end/{end}/{readingCount} methods=[GET] msg="Route added"
level=INFO ts=2021-10-31T22:32:31.722122764Z app=datastorage source=restrouter.go:75 msg="Registering v2 routes..."
level=DEBUG ts=2021-10-31T22:32:31.72264084Z app=datastorage source=profiles.go:47 msg="created absolute path for loading pre-defined Device Profiles: /edge-orchestration/res"
level=DEBUG ts=2021-10-31T22:32:31.724352674Z app=datastorage source=devices.go:29 msg="Loading pre-define Devices from configuration"
level=DEBUG ts=2021-10-31T22:32:31.724387491Z app=datastorage source=devices.go:32 msg="Device edge-orchestration-c1b23cc6-0767-400a-9cf0-36e1b3902da2 exists, using the existing one"
```

Comment (Seunghui98) [reply internal/controller/storagemgr/storage.go]: @t25kim 
Thanks for your advice. :D
As advised, the location of deviceName was modified into StartStorage function, and the test result came out as above. Please review the code again.

**Review by t25kim — APPROVED** [review]
LGTM Thanks for contribution!

**Comment by tdrozdovsky** [comment]
LGTM

**Review by tdrozdovsky — APPROVED** [review]
LGTM

**Review by MoonkiHong — APPROVED** [review]
+1

**Pull request merged**